### PR TITLE
Enable cppcoreguidelines-special-member-functions

### PR DESCRIPTION
**Pull request by masterleinad**
This would have caught #368. Needs some resolution of #368 and #371.

### CODE REVIEW

File: benchmarks/bvh_driver/bvh_driver.cpp [hunk 1]
```diff
@@ -233,6 +233,10 @@ class KokkosScopeGuard
 {
 public:
   KokkosScopeGuard(int &argc, char *argv[]) { Kokkos::initialize(argc, argv); }
+  KokkosScopeGuard(const KokkosScopeGuard &) = default;
+  KokkosScopeGuard(KokkosScopeGuard &&) = default;
+  KokkosScopeGuard &operator=(const KokkosScopeGuard &) = default;
+  KokkosScopeGuard &operator=(KokkosScopeGuard &&) = default;
   ~KokkosScopeGuard() { Kokkos::finalize(); }
 };
```
Comment (aprokop) [inline benchmarks/bvh_driver/bvh_driver.cpp:242]: This one should be gone in the first place, as there is `Kokkos::ScopeGuard`.

**Review by aprokop — APPROVED** [review]
I have no real opinion on this. Most of the caught stuff is fluff guards in tests, that we really don't do anything with. I would be happier if we just run this tool on `src/`.

**Review by aprokop — COMMENTED** [review]
Did not mean to approve, just wanted to comment.

**Review by dalg24 — COMMENTED** [review]
I am not really inclined to add that check but I won't block it if you guys decide you really want it.

**Comment by aprokop** [comment]
retest this please

**Comment by aprokop** [comment]
retest this please

**Comment by aprokop** [comment]
I've got nothing to say for or against this PR, as I am not that well versed in constructors and ramifications of this. I'd say merge it if there is no downside.

**Comment by masterleinad** [comment]
Seems no one except for me wants this, so closing.

**Pull request closed without merging**